pkg/controller/priorityqueue/priorityqueue.go

```diff
@@ -326,36 +331,64 @@ func (w *priorityqueue[T]) Len() int {
 	return result
 }
 
+func (w *priorityqueue[T]) logState() {
+	t := time.Tick(10 * time.Second)
+	for {
+		select {
+		case <-w.done:
+			return
+		case <-t:
+		}
+
+		// Log level may change at runtime, so keep the
+		// loop going even if a given level is currently
+		// not enabled.
+		if !w.log.V(5).Enabled() {
+			continue
+		}
+		items := make([]*item[T], 0, len(w.items))
+		w.lock.Lock()
+		w.queue.Ascend(func(item *item[T]) bool {
+			items = append(items, item)
+			return true
+		})
+		w.lock.Unlock()
+
+		w.log.V(5).Info("workqueue_state", "items", items)
```

Comment:
```suggestion
		w.log.V(5).Info("Workqueue items", "items", items)
```
Maybe this makes it look a bit more like a regular log message. No strong opinion, up to you.

Reply:
I don't like the space in there because that makes the query more annoying with some log query systems and I think its uncommon to uppercase, so I ended up with `workqueue_items` - WDYT?

Reply:
Sounds good